src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,210 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = new Node<>(null, null, null);
+        tail = new Node<>(null, null, null);
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size == 0) {
+            addFirstNode(value, newNode);
+        } else {
+            tail.setNext(newNode);
+            newNode.setPrev(tail);
+            tail = newNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0 && size == 0) {
+            addFirstNode(value, newNode);
+        } else {
+            if (index == 0) {
+                newNode.setNext(head);
+                newNode.setPrev(null);
+                head.setPrev(newNode);
+                head = newNode;
+                size++;
+            } else if (index == size) {
+                newNode.setNext(null);
+                newNode.setPrev(tail);
+                tail.setNext(newNode);
+                tail = newNode;
+                size++;
```

Comment:
you can move this if statement to top of the method
```suggestion
            } else if (index == size) {
                // you can use add(T value) here
```